Competition/src/main/cpp/subsystems/Feeder.cpp

```diff
@@ -72,70 +71,117 @@ void Feeder::assessInputs()
     state.operator_leftBumperPressed = operatorController->GetLeftBumper();
     
 
-    state.lowerBannerTripped = false;//banner_upper.Get();
-    state.upperBannerTripped = false;//banner_upper.Get();
+    state.bannerTripped = !banner.Get();
 
-    if (state.driver_rightBumperPressed || state.operator_leftBumperPressed) {
-        state.feederState = FeederState::FEEDER_INTAKE2;
+    state.currentBanner = state.bannerTripped;
+    
+    if (state.operator_leftBumperPressed) {
+        state.feederState = FeederState::FEEDER_SHOOT;
+        state.spiked = false;
     }
     else if (state.operator_bButtonPressed || state.driver_leftBumperPressed) {
         state.feederState = FeederState::FEEDER_REVERSE;
+        state.spiked = false;
     }
-    else if (state.operator_aButtonPressed) {
-        // if (state.lowerBannerTripped) {
-        //     state.feederState = state.upperBannerTripped ? FeederState::FEEDER_DISABLE : FeederState::FEEDER_INTAKE2;
-        // } 
-        // else if (state.upperBannerTripped) {
-        //     state.feederState = FeederState::FEEDER_INTAKE1;
-        // }
-        // else {
-            state.feederState = FeederState::FEEDER_INTAKE1;
-       // }
+    else if (state.operator_aButtonPressed || state.driver_rightBumperPressed) {
+        if (state.bannerTripped) {
```

Comment:
Logic needs to be moved to assignOutputs or analyzeDashboard in order for it to work in Auto

Reply:
made new auto_intake state and moved it to assign outputs